jetty-io/src/main/java/org/eclipse/jetty/io/ManagedSelector.java

```diff
@@ -144,6 +148,36 @@ protected void doStop() throws Exception
         super.doStop();
     }
 
+    @ManagedAttribute(value = "Total number of keys", readonly = true)
+    public int getTotalKeys()
+    {
+        return _selector.keys().size();
+    }
+
+    @ManagedAttribute(value = "Average number of selected keys", readonly = true)
+    public double getAverageSelectedKeys()
+    {
+        return _keyStats.getMean();
+    }
+
+    @ManagedAttribute(value = "Maximum number of selected keys", readonly = true)
+    public double getMaxSelectedKeys()
+    {
+        return _keyStats.getMax();
+    }
+
+    @ManagedAttribute(value = "Total number of select() calls", readonly = true)
+    public long getSelectCount()
+    {
+        return _keyStats.getCount();
+    }
+
+    @ManagedOperation(value = "Resets the statistics", impact = "ACTION")
+    public void resetStats()
```

Comment:
Are you _really sure_ we want to expose this?

Reply:
I think it is good to have and easy to ignore if you don't want to use it.